scvid/data/distributed_anndata.py

```diff
@@ -301,7 +302,14 @@ def adata(self) -> AnnData:
             )
             self.schema.validate_anndata(adata)
             # cache anndata
-            self.cache[self.filename] = adata
+            if len(self.cache) < self.cache.max_size:
+                self.cache[self.filename] = adata
+            else:
+                self.cache[self.filename] = adata
```

Comment:
This looks alright to me, but I'd suggest having an eye on memory leak in real usage (training on lots of AnnData) in case the `gc.collect()` trick is not sufficient. There were other tricks, too (e.g. replacing `.X`. with an empty sparse matrix _before_ calling collect() -- sounds magical to me but the guy in the scverse issue claimed it did the trick).

Reply:
I'll keep an eye on it. If there is a problem it should quickly become apparent once we start analyzing much larger datasets. I just don't want to introduce unnecessary fixes before an actual issue arises.